src/components/Bar/Bar.component.jsx

```diff
@@ -0,0 +1,127 @@
+import Toolbar from '@material-ui/core/Toolbar';
+import IconButton from '@material-ui/core/IconButton';
+import Typography from '@material-ui/core/Typography';
+import InputBase from '@material-ui/core/InputBase';
+import { fade, makeStyles } from '@material-ui/core/styles';
+import MenuIcon from '@material-ui/icons/Menu';
+import Button from '@material-ui/core/Button';
+import SearchIcon from '@material-ui/icons/Search';
+import MyContext from '../../MyContext'
+import LogInAvatar from '../Avatar'
+import { useHistory } from 'react-router-dom'
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    flexGrow: 3,
+  },
+  menuButton: {
+    marginRight: theme.spacing(1),
+  },
+  title: {
+    flexGrow: 1,
+    display: 'none',
+    [theme.breakpoints.up('sm')]: {
+      display: 'block',
+    },
+  },
+  search: {
+    position: 'relative',
+    borderRadius: theme.shape.borderRadius,
+    backgroundColor: fade(theme.palette.common.white, 0.15),
+    '&:hover': {
+      backgroundColor: fade(theme.palette.common.white, 0.25),
+    },
+    marginLeft: 0,
+    width: '100%',
+    [theme.breakpoints.up('sm')]: {
+      marginLeft: theme.spacing(1),
+      width: 'auto',
+    },
+  },
+  searchIcon: {
+    padding: theme.spacing(0, 2),
+    height: '100%',
+    position: 'absolute',
+    pointerEvents: 'none',
+    display: 'flex',
+    alignItems: 'center',
+    justifyContent: 'center',
+  },
+  inputRoot: {
+    color: 'inherit',
+  },
+  inputInput: {
+    padding: theme.spacing(1, 1, 1, 0),
+    paddingLeft: `calc(1em + ${theme.spacing(4)}px)`,
+    transition: theme.transitions.create('width'),
+    width: '100%',
+    [theme.breakpoints.up('sm')]: {
+      width: '12ch',
+      '&:focus': {
+        width: '20ch',
+      },
+    },
+  },
+}));
+
+function Bar(props) {
+  const classes = useStyles();
+  const context = useContext(MyContext)
+  const history =useHistory()
+
+
+const GoFavorites= () => {
+    history.push(`/favorites`)
+}
+
+const GoHome= () => {
+  history.push(`/`)
+}
+
+
+  return (
+    <div className={classes.root}>
+
+      <AppBar position="static">
+
+        <Toolbar>
+          <IconButton
+            edge="start"
+            className={classes.menuButton}
+            color="inherit"
+            aria-label="open drawer"
+          >
+            <MenuIcon />
+          </IconButton>
+          <Typography className={classes.title} variant="h6" noWrap>
+            React Proyect Video
+          </Typography>
+          <Button onClick={GoFavorites} variant="contained" color="primary" href="#contained-buttons">
+            Favorites
+          </Button>
+          <Button onClick={GoHome} variant="contained" color="primary" href="#contained-buttons">
+            Home
+          </Button>
+          <div className={classes.search}>
+            <div className={classes.searchIcon}>
+              <SearchIcon />
+            </div>
+            <InputBase onChange= {(e)=> context.searchVideos(e.target.value)}
```

Comment:
I suggest you throttle the search functionality, this applies specially if searchVideos becomes a network requests but even for in memory linear searches this is necessary.

Reply:
Just to clarify:

Throttling is a technique in which, no matter how many times the user fires the event, the attached function will be executed only once in a given time interval.